CHANGELOG.md

```diff
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Features
 ### Enhancements
 ### Bug Fixes
+- Fixed merge logic for multiple collector result case ([#800](https://github.com/opensearch-project/neural-search/pull/800)) 
```

Comment:
A better changelog would be "Fix for HybridQuery results are missing when concurrent segment search is enabled"

Reply:
ack

Reply:
I cannot see the fix here.

Reply:
corrected